### PR TITLE
[Enhance] Removed augmentation functional module.

### PR DESCRIPTION
**Pull request by shijianjian**
Removed augmentation functional module and the corresponding tests and docs.

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

**Comment by cceyda** [comment]
maybe we should move the deprecation logic in  `kornia/augmentation/functional/__temp__.py` to `kornia/utils/deprecate.py` instead of deleting it. we can always use it later 😉 

Do we have something similar to `apply_erase_rectangles` once it is deprecated? it feels useful 🤔 https://kornia.readthedocs.io/en/latest/search.html?q=erase&check_keywords=yes&area=default

**Comment by edgarriba** [comment]
Another option is to move the code under kornia/graveyard

**Comment by shijianjian** [comment]
> Do we have something similar to `apply_erase_rectangles` once it is deprecated? it feels useful 🤔 https://kornia.readthedocs.io/en/latest/search.html?q=erase&check_keywords=yes&area=default
> 
It is more or less proposed as a data augmentation technique, and it can be simply achieved by ~~~bbox_to_mask~~~ then a multiplication.

**Comment by shijianjian** [comment]
> Another option is to move the code under kornia/graveyard

Cannot see the point to maintain the code that we decide not to maintain.

**Comment by justanhduc** [comment]
Sorry I didn't update on this. Why did we decide to remove this module? But I agree that there's no point to keep it without maintaining. The API keeps changing alot and it will go out of the Kornia's standard soon.

**Comment by shijianjian** [comment]
> Sorry I didn't update on this. Why did we decide to remove this module? But I agree that there's no point to keep it without maintaining. The API keeps changing alot and it will go out of the Kornia's standard soon. 

We had too many functional calls for the previous implementation, so that users needed to jump from files to files. That's basically the reason.

File: kornia/augmentation/functional/__temp__.py [hunk 1]
```diff
@@ -1,24 +0,0 @@
-from typing import Callable
```
Comment (justanhduc) [inline kornia/augmentation/functional/__temp__.py:1]: This can be reused later. I think that what @cceyda meant in the previous comment.

**Comment by edgarriba** [comment]
Any final decision here ?

**Comment by edgarriba** [comment]
@shijianjian could you rebase - too many changes this days

**Pull request merged**